catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java

```diff
@@ -511,6 +534,24 @@ private QueryRequest validateQueryRequest(QueryRequest queryRequest)
                     "Cannot perform query with null query, either passed in from endpoint, or as output from a PreQuery Plugin");
         }
 
+        Query originalQuery = queryRequest.getQuery();
```

Comment:
❓ Instead of re-assigning `queryRequest` and putting everything in the `if` block, should we do something like this?

```
if ( queryPageSize >= 1 && queryPageSize <= MAX_PAGE_SIZE ) {
    return queryRequest;
}

Query originalQuery = queryRequest.getQuery();

Query modifiedQuery = new QueryImpl(originalQuery,
        originalQuery.getStartIndex(),
        MAX_PAGE_SIZE,
        originalQuery.getSortBy(),
        originalQuery.requestsTotalResultsCount(),
        originalQuery.getTimeoutMillis());

return new QueryRequestImpl(modifiedQuery,
        queryRequest.isEnterprise(),
        queryRequest.getSourceIds(),
        queryRequest.getProperties());
```